finesse/hardware/temperature/dummy_temperature_monitor.py

```diff
@@ -0,0 +1,43 @@
+"""This module provides an interface to dummy DP9800 temperature readers."""
+import logging
+from datetime import datetime
+from decimal import Decimal
+
+from pubsub import pub
+
+from ..noise_producer import NoiseProducer
+from .temperature_monitor_base import TemperatureMonitorBase
+
+
+class DummyTemperatureMonitor(TemperatureMonitorBase):
+    """A dummy temperature monitor for GUI testing."""
+
+    def __init__(self) -> None:
+        """Create a new DummyTemperatureMonitor."""
+        super().__init__("dummy")
+
+    def close(self) -> None:
+        """Close the connection to the device."""
+        pub.sendMessage("temperature_monitor.close")
+        logging.info("Closed connection to dummy temperature monitor")
+
+    def send_temperatures(self) -> None:
+        """Publish fluctuating temperatures."""
+        noise = NoiseProducer(seed=datetime.now().second)()
+
+        time_now = datetime.now().timestamp()
+        temperatures = [
+            Decimal(19 + noise),
+            Decimal(17 + noise),
+            Decimal(26 + noise),
+            Decimal(850.00),
+            Decimal(24 + noise),
+            Decimal(68 + noise),
+            Decimal(69 + noise),
+            Decimal(24 + noise),
+        ]
```

Comment:
A couple of things:

1. You don't need to worry about setting a random seed (I used a fixed seed on purpose)
2. This currently uses the same random number for each channel

How about something like this:

```suggestion
        _BASE_TEMPS = (19, 17, 26, 850, 24, 68, 69)
        rand = NoiseProducer()

        time_now = datetime.now().timestamp()
        temperatures = [rand() + temp for temp in _BASE_TEMPS]
```

Reply:
Ahh my bad. So it will. In that case I'd make the `NoiseProducer` an attribute of the class. Setting up random number generators is a little slow so you don't want to do it more often than you have to. It won't be a bottleneck here, but it's a good habit to get into.